### PR TITLE
Hexgold Slash

### PR DESCRIPTION
**Pull request by AhmadYProjects**
haven't tested but used Burn the Impure as a reference when writing this code as it is similar enough. I don't really know how to modify ToxicAbility so that i can apply getInstance onto it, so I'll just leave that to someone more knowledgeable.

### CODE REVIEW

**Comment by AhmadYProjects** [comment]
[[Hexgold Slash]]

File: Mage.Sets/src/mage/cards/h/HexgoldSlash.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    @Override
+    public HexgoldSlashEffect copy(){
+        return new HexgoldSlashEffect();
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source){
+        Permanent permanent = game.getPermanent(getTargetPointer().getFirst(game, source));
+        if (permanent != null){
+            if (permanent.getAbilities().(ToxicAbility.getInstance())) {
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/h/HexgoldSlash.java:60]: use `permanent.getAbilities(game).containsClass(ToxicAbility.class)`

File: Mage.Sets/src/mage/cards/h/HexgoldSlash.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                permanent.damage(2,source,game);
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/h/HexgoldSlash.java:61]: don't forget to put 4 here

File: Mage.Sets/src/mage/cards/h/HexgoldSlash.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    @Override
+    public HexgoldSlash copy() {
+        return new HexgoldSlash(this);
+    }
+}
+
+class HexgoldSlashEffect extends OneShotEffect{
+    public HexgoldSlashEffect(){
+        super(Outcome.Damage);
+        staticText = "{this} deals 2 damage to target creature. If that creature has toxic, {this} deals 4 damage instead}";
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/h/HexgoldSlash.java:44]: `}` at the end of the string

**Review by theelk801 — CHANGES_REQUESTED** [review]

**Comment by theelk801** [comment]
try to write descriptive commit messsages

File: Mage.Sets/src/mage/cards/h/HexgoldSlash.java (hunk 3 above)
Comment (AhmadYProjects) [reply Mage.Sets/src/mage/cards/h/HexgoldSlash.java]: > try to write descriptive commit messages

Ok I will try to do that from here on out. all requested changes have been implemented in the latest commit

Comment (theelk801) [reply Mage.Sets/src/mage/cards/h/HexgoldSlash.java]: thanks, looks good to me

**Pull request merged**